CMSgov/mdct-mcr
# [MDCT-1959] Clear nested fields after they are hidden from view

## Description
Closes [MDCT-1959: Nested fields shouldn't persist data in form state or save to database](https://qmacbis.atlassian.net/browse/MDCT-1959)
Also fixes what I assume were some incorrect validation types and removes a comma-separated mask on a textarea.
Note: this doesn't really work for fields that are already persisted to the database. Don't worry, it's fine. We'll address that when we do autosave 😬 

### How to test
* After filling out a nested field, unselect (for a checkbox) or change the parent value (radio) to hide the nested field. Then reselect the original parent value and ensure the nested field does not display what you entered previously.
* For textarea nested under checkbox: Section B, Encounter Data Report
* For textarea nested under radio: Section B, Program Integrity
* For radio nested under checkbox: Section B, Encounter Data Report
* For number nested under radio: Section D, Program Integrity
* For date nested under radio: Section D, Quality Measures

### Changed Dependencies
N/A

## Code author checklist

## Reviewer checklist (two different people)
